relay-general/src/store/normalize.rs

```diff
@@ -1001,6 +1002,78 @@ impl<'a> Processor for NormalizeProcessor<'a> {
         contexts.0.remove("reprocessing");
         Ok(())
     }
+
+    fn process_event_logger(
+        &mut self,
+        event_logger: &mut EventLogger,
+        meta: &mut Meta,
+        state: &ProcessingState,
+    ) -> ProcessingResult {
+        event_logger.process_child_values(self, state)?;
+
+        if event_logger.is_empty() {
+            return Ok(());
+        }
+
+        let trimmed_logger = event_logger.as_str().trim();
+        let logger_len = bytecount::num_chars(trimmed_logger.as_bytes());
+
+        if trimmed_logger.is_empty() {
+            return Err(ProcessingAction::DeleteValueHard);
+        } else if logger_len > MaxChars::Logger.limit() {
+            meta.add_error(Error::new(ErrorKind::ValueTooLong));
+            meta.set_original_length(Some(logger_len));
+
+            let new_logger = shorten_logger(trimmed_logger);
+            event_logger.set_logger(new_logger);
+        } else if trimmed_logger != event_logger.as_str() {
+            event_logger.set_logger(trimmed_logger.to_owned());
+        }
+
+        Ok(())
+    }
+}
+
+static LOGGER_DELIMITER_PATTERN: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"\.").expect("Invalid logger separator Regex"));
+static LOGGER_TRIMMED_PREFIX: &str = "*";
+
+fn shorten_logger(logger: &str) -> String {
+    let mut tokens = LOGGER_DELIMITER_PATTERN.split(logger).collect_vec();
+    tokens.reverse(); // Prioritize keeping chars from the end
+
+    let logger_len = bytecount::num_chars(logger.as_bytes());
+    let target_size = logger_target_size(tokens);
+    let target_idx = logger_len - target_size;
+
+    LOGGER_TRIMMED_PREFIX.to_owned() + &logger[target_idx..]
+}
+
+fn logger_target_size(tokens: Vec<&str>) -> usize {
+    let logger_prefix_len = bytecount::num_chars(LOGGER_TRIMMED_PREFIX.as_bytes());
+    if MaxChars::Logger.limit() < logger_prefix_len {
+        return 0;
+    }
+
+    let max = MaxChars::Logger.limit() - logger_prefix_len;
+    let mut cummulative = 0;
+
+    for token in tokens {
+        // Consider one slot for the delimiter between tokens.
+        let current = bytecount::num_chars(token.as_bytes()) + 1;
```

Comment:
It's a good idea to use `num_chars` for limiting the string, but the computed `target_size` that is used for slicing the string should be in bytes. Best add some unit tests with strings that contain a lot of special characters to see the effects.